Feature01

hasi-b opened the pull request:
pull request to review

hasi-b committed:
initial commit

hasi-b committed:
player movement implemented

hasi-b committed:
lookAt mouse pointer implemented

hasi-b committed:
minor fix of camera in cinemachine

Added an object which the cinemachine camera can hard lock in to

hasi-b committed:
bullet types added

Two types of bullet added

hasi-b committed:
shoot implemented

hasi-b committed:
movement and shooting implemented

hasi-b commented on Assets/Feature01/Scripts/Bullet/GreenBullet.cs:
```diff
@@ -0,0 +1,14 @@
+
+public class GreenBullet : BulletAbstractClasses
+{
+  
```
Color should be a property.

hasi-b commented on Assets/Feature01/Scripts/Controller/Instantiation.cs:
```diff
@@ -0,0 +1,13 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Instantiation : MonoBehaviour
```
Enum class should not be a mono behaviour

hasi-b commented on Assets/Feature01/Scripts/Controller/PlayerController.cs:
```diff
@@ -0,0 +1,61 @@
+{
+    Rigidbody rb;
+    [SerializeField] PlayerOnFootInput PlayerFootInput;
+    [SerializeField] PlayerInput pl;
```
write fully qualified name for properties / variables

hasi-b commented on Assets/Feature01/Scripts/Controller/PlayerController.cs:
```diff
@@ -0,0 +1,61 @@
+        
+        PlayerMove();
+        PlayerLook();
+        rb.AddForce(playerMoveInput,ForceMode.Force);
```
fully qualified name

hasi-b commented on Assets/Feature01/Scripts/Controller/Shoot/ShootSystem.cs:
```diff
@@ -0,0 +1,38 @@
+        playerFootInput = GetComponent<PlayerOnFootInput>();
+    }
+
+    public  void playerShoot(Vector3 mousePosition)
```
check if this is correct implementation with New Input System

hasi-b commented on Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs:
```diff
@@ -0,0 +1,78 @@
+    public Vector3 getMousePosition()
+    {  
+        
+        return mainCamera.ScreenToWorldPoint(new Vector3(Input.mousePosition.x, Input.mousePosition.y, mainCamera.transform.position.y));
```
Use new inputsystem only

hasi-b reviewed (commented):
Need to update according to the comnments

hasi-b committed:
Changes made based on the comments

hasi-b committed:
gitignore changed for meta

hasi-b committed:
changed based on the comment

hasi-b committed:
Player movement keyboared inmplemented

hasi-b committed:
Player Movement & Shoot at Gamepad right stick direction

Warhammer4000 commented on Assets/Feature01/Scripts/AbstractClasses/CharacterAbstractClass.cs:
```diff
@@ -0,0 +1,8 @@
+
+public abstract class CharacterAbstractClass
+{
+    public abstract void characterInstantiate();
```
Method Name will be CamelCase. 
First character will be capital. Need to follow one Standard so that everyone is on the same ground

hasi-b replied:
yes bhai update kortesi

Warhammer4000 commented on Assets/Feature01/Scripts/Characters/Enemy.cs:
```diff
@@ -0,0 +1,11 @@
+
+public class Enemy : CharacterAbstractClass
+{
+    public override void characterInstantiate()
```
Capital

Warhammer4000 commented on Assets/Feature01/Scripts/Characters/Player.cs:
```diff
@@ -0,0 +1,13 @@
+public class Player : CharacterAbstractClass
+{
+
+    public override void characterInstantiate()
```
Capital

Warhammer4000 commented on Assets/Feature01/Scripts/Controller/MovementController.cs:
```diff
@@ -0,0 +1,50 @@
+    Vector3 playerMoveInput;
+
+    public void PlayerMove(Vector2 movementInput, float speed,Rigidbody _rigidbody)
+    {
```
Are you sure you wish to create movement based on rigid body? 
Open ended question. It can be something really fun.. Or something very complicated to manage.

Warhammer4000 commented on Assets/Feature01/Scripts/Controller/PlayerController.cs:
```diff
@@ -0,0 +1,84 @@
+    }
+
+
+    private void FixedUpdate()
```
This is not how you should use New input system. You are basically using the same paradigm as old input system. I recommend watching a few tutorials.

How it should work is . whenever your stick is moving something is happening. Event driven. Not Update Driven.

hasi-b replied:
bhai ami kiso tutorial dekhsilam,  oigula te emon fixed update er vitor  diye kora, event driven input  niye constantly check kore. 
bhai any recommendation on tutorial jekhane only event fire korle e move hocche

Warhammer4000 commented on Assets/Feature01/Scripts/Input/GamePadCursor.cs:
```diff
@@ -0,0 +1,47 @@
+
+
+    public void moveCursorPosition(Vector2 cursorMovementInput,float _cursorSpeed)
+    {
```
Why did you need this?
You should have a look at the cursor class on unity API.

hasi-b replied:
eta gamepad diye jokhn first e cursor movement  implement korechilam tokhn kora, baad diye dicchi

Warhammer4000 commented on Assets/Feature01/Scripts/Input/GamePadCursor.cs:
```diff
@@ -0,0 +1,47 @@
+using UnityEngine.InputSystem.LowLevel;
+using UnityEngine.InputSystem.Users;
+
+public class GamePadCursor : MonoBehaviour
```
Gamepad cursor is not needed. Just use the direction vector of the Right Axis. Shoot at the direction.

hasi-b replied:
ji bhai baad ditesi eta

Warhammer4000 commented on Assets/Feature01/Scripts/Input/PlayerOnFootInput.cs:
```diff
@@ -0,0 +1,106 @@
+
+   
+
+    public Vector3 getMousePosition()
```
You probably don't need this. Not sure. Do you? Can you explain briefly.

hasi-b replied:
yes bhai, jokhn mouse based shooting implement korsilam first e tokhn kora, pore baad deya hoy naai, comment out kore dicchi

Warhammer4000 reviewed (changes requested):
Have a look

hasi-b committed:
Addressed the commented issues

Fixed the commented issues of the pull request, accept taking input on updates and moving the player with rigidbody

hasi-b committed:
Update default-2021.dwlt

hasi-b committed:
Dash mechanics test

hasi-b committed:
POL-23

minor change in controller, using _rigidbody velocity instead of Add force

hasi-b committed:
POL-23

Dash mechanics added

hasi-b committed:
changed namings
